packages/react-components/react-popover/src/components/Popover/usePopover.ts

```diff
@@ -57,7 +57,33 @@ export const usePopover_unstable = (props: PopoverProps): PopoverState => {
     popoverSurface = children[0];
   }
 
-  const [open, setOpen] = useOpenState(initialState);
+  const [open, setOpenState] = useOpenState(initialState);
+
+  const setOpenTimeoutRef = React.useRef(0);
+
+  const setOpen = useEventCallback((e: OpenPopoverEvents, shouldOpen: boolean) => {
+    clearTimeout(setOpenTimeoutRef.current);
+
+    if (e.type === 'mouseleave') {
+      // FIXME leaking Node timeout type
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      setOpenTimeoutRef.current = setTimeout(() => {
+        setOpen(e, shouldOpen);
```

Comment:
I'm pretty sure this will cause an infinite timeout loop

1. call setOpen with mouseleave event
2. timeout calls setOpen with the same mouseleave event
3. ...

Reply:
Oops this is a terrible typo 🙈